### PR TITLE
Add Matomo and Sentry features optout

### PR DESCRIPTION
**Pull request by NicolasBourdin88**

### CODE REVIEW

File: app/src/main/res/layout/fragment_management_data_settings.xml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+        <com.infomaniak.mail.ui.main.settings.ItemSettingView
+            android:id="@+id/dataManagementSentry"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            app:icon="@drawable/ic_sentry"
+            app:itemAction="chevron"
+            app:title="@string/settingsSentryTitle" />
+    </LinearLayout>
+</com.infomaniak.mail.ui.main.menu.SimpleSettingView>
```
Comment (KevinBoulongne) [inline app/src/main/res/layout/fragment_management_data_settings.xml:82]: There's probably a missing setting in your Android Studio that needs to be enabled to add empty lines at end of files.

File: app/src/main/java/com/infomaniak/mail/MainApplication.kt [hunk 2]
```diff
@@ -72,7 +72,7 @@ import javax.inject.Inject
 @HiltAndroidApp
 open class MainApplication : Application(), ImageLoaderFactory, DefaultLifecycleObserver, Configuration.Provider {
 
-    val matomoTracker: Tracker by lazy { buildTracker() }
+    val matomoTracker: Tracker by lazy { buildTracker(!localSettings.isMatomoTrackingEnabled) }
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/MainApplication.kt:75]: Add named parameter.

File: app/src/main/res/drawable-night/ic_matomo.xml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+<!--
```
Comment (KevinBoulongne) [inline app/src/main/res/drawable-night/ic_matomo.xml:1]:
```suggestion
<?xml version="1.0" encoding="utf-8"?><!--
```

File: app/src/main/res/drawable-night/ic_sentry_text.xml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (KevinBoulongne) [inline app/src/main/res/drawable-night/ic_sentry_text.xml:1]: Missing header, check everywhere.

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

**Review by LunarX — CHANGES_REQUESTED** [review]

**Review by sirambd — APPROVED** [review]

File: app/src/main/java/com/infomaniak/mail/MainApplication.kt [hunk 5]
```diff
@@ -166,9 +166,10 @@ open class MainApplication : Application(), ImageLoaderFactory, DefaultLifecycle
     private fun configureSentry() {
         SentryAndroid.init(this) { options: SentryAndroidOptions ->
             // Register the callback as an option
-            options.beforeSend = SentryOptions.BeforeSendCallback { event: SentryEvent?, _: Any? ->
+            options.beforeSend = SentryOptions.BeforeSendCallback { event: SentryEvent, _: Any? ->
                 // If the application is in debug mode, discard the events
-                if (BuildConfig.DEBUG) null else event
+                val isRelease = !BuildConfig.DEBUG
+                if (localSettings.isSentryTrackingEnabled && isRelease) event else null
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/MainApplication.kt:172]:
```suggestion
                if (!BuildConfig.DEBUG && localSettings.isSentryTrackingEnabled) event else null
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

**Review by LunarX — APPROVED** [review]

File: app/src/main/res/navigation/main_navigation.xml [hunk 6]
```diff
@@ -306,6 +306,9 @@
         <action
             android:id="@+id/action_settingsFragment_to_aiEngineSettingFragment"
             app:destination="@id/aiEngineSettingFragment" />
+        <action
+            android:id="@+id/action_settings_to_managementDataSetting"
```
Comment (FabianDevel) [inline app/src/main/res/navigation/main_navigation.xml]:
```suggestion
            android:id="@+id/action_settings_to_dataManagementSettings"
```

**Review by KevinBoulongne — APPROVED** [review]

File: app/src/main/res/navigation/main_navigation.xml [hunk 7]
```diff
@@ -427,6 +430,31 @@
...
+    <fragment
+        android:id="@+id/dataManagementSettingsFragment"
+        android:name="com.infomaniak.mail.ui.main.settings.privacy.DataManagementSettingsFragment"
+        android:label="DataManagementSettingsFragment"
+        tools:layout="@layout/fragment_data_management_settings">
+        <action
+            android:id="@+id/action_dataManagement_to_matomoSetting"
+            app:destination="@id/dataManagementMatomoSettingFragment" />
+        <action
+            android:id="@+id/action_dataManagement_to_sentrySetting"
```
Comment (FabianDevel) [inline app/src/main/res/navigation/main_navigation.xml:442]:
```suggestion
            android:id="@+id/action_dataManagementSettingsFragment_to_dataManagementSentrySettingFragment"
```

**Review by FabianDevel — CHANGES_REQUESTED** [review]

**Pull request merged**